File: src/main/java/pl/allegro/experiments/chi/chiserver/domain/experiments/DeviceClassPredicate.java
package pl.allegro.experiments.chi.chiserver.domain.experiments;

import com.google.common.base.Preconditions;
import com.google.common.base.Strings;

public class DeviceClassPredicate implements Predicate {
    private final String device;

    public DeviceClassPredicate(String device) {
        Preconditions.checkArgument(!Strings.isNullOrEmpty(device));
        this.device = device.toLowerCase();
    }

    public String getDevice() {
        return device;
    }
}

File: src/main/kotlin/pl/allegro/experiments/chi/chiserver/domain/DeviceClassPredicate.kt
package pl.allegro.experiments.chi.chiserver.domain

import com.google.common.base.Preconditions
import com.google.common.base.Strings

class DeviceClassPredicate(device: String) : Predicate {
    val device: String

    init {
        Preconditions.checkArgument(!Strings.isNullOrEmpty(device), "empty device in DeviceClassPredicate")

        this.device = device.toLowerCase()
    }
}
